controllers/machinehealthcheck_targets.go

```diff
@@ -135,14 +135,18 @@ func (t *healthCheckTarget) needsRemediation(logger logr.Logger, timeoutForMachi
 		}
 		logger.V(3).Info("Using comparison time", "time", comparisonTime)
 
-		if comparisonTime.Add(timeoutForMachineToHaveNode).Before(now) {
+		if timeoutForMachineToHaveNode != time.Duration(0) && comparisonTime.Add(timeoutForMachineToHaveNode).Before(now) {
```

Comment:
given that we are using the disable value in a couple places, should we make it a const somewhere?

Reply:
Could we use `!timeoutForMachineToHaveNode.IsZero()` instead?

Reply:
Durations don't have an `IsZero`, but I've made a little helper and moved the logic earlier in the if block which I think makes things a bit more obvious